### PR TITLE
Webpack

### LINKED ISSUE #524: Move to WebPack
Building the frontend is annoying, and updating dependencies in a way that lets us test the updates is nothing short of tedious.

WebPack with proper bounds on the minor version of each component and dependency should help us stay in line and keep up to date with the frontend code.

### PR DESCRIPTION
**Pull request by anirudhpillai**
## What does this PR do?
Moves from Gulp to webpack.

Using `webpack-serve` for local development now.
If `UCLAPI_PRODUCTION` in .env is set to False then the static files are served from `webpack-serve`.

This resolves #524

### CODE REVIEW

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/1426/
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/1427/
Jenkins statics successfully built.

File: backend/uclapi/dashboard/templates/getStarted.html [hunk 1]
```diff
@@ -20,7 +19,11 @@
     window.staticURL = "{% static "" %}";
   </script>
   <div class="app"></div>
-  <script src="{% static "js/getStarted.js" %}"></script>
+  {% if WEBPACK_SERVE %}
```
Comment (ChristopherHammond13) [inline backend/uclapi/dashboard/templates/getStarted.html]: Is there a less hacky solution to this? Perhaps set the static path to http://localhost:8080 using an environment variable?

File: frontend/package.json [hunk 2]
```diff
@@ -51,10 +53,19 @@
     "gulp-strip-debug": "^3.0.0",
```
Comment (ChristopherHammond13) [inline frontend/package.json]: Do we still need the `gulp` dependencies?

File: backend/uclapi/dashboard/templates/getStarted.html (hunk 1 above)
Comment (anirudhpillai) [reply backend/uclapi/dashboard/templates/getStarted.html]: that's what I was doing but webpack-serve (http://localhost:8080) doesn't serve images and for that we had to go back to the Django static file server

File: frontend/package.json (hunk 2 above)
Comment (anirudhpillai) [reply frontend/package.json]: nice catch, I'll remove those

File: backend/uclapi/dashboard/templates/getStarted.html (hunk 1 above)
Comment (ChristopherHammond13) [reply backend/uclapi/dashboard/templates/getStarted.html]: Ahhh I see. Is this an inbuilt restriction in Webpack? It seems strange that it wouldn't serve the files. Does this explain it at all? https://stackoverflow.com/questions/34870888/serving-static-assets-in-webpack-dev-server

Full disclosure: I know literally nothing about Webpack and very little about frontend so there might be something strange about our setup that prevents this from happening.

File: frontend/package.json (hunk 2 above)
Comment (ChristopherHammond13) [reply frontend/package.json]: 💖

File: backend/uclapi/dashboard/templates/getStarted.html (hunk 1 above)
Comment (anirudhpillai) [reply backend/uclapi/dashboard/templates/getStarted.html]: I did think of that, but I wasn't sure how it'll affect the serving of images as now the entire images folder will now be moved to `frontend/`. It's a lot of work to implement it as I'll need to change the code for images on the frontend, but I'll still give it a try.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/1467/
Jenkins statics successfully built.

File: backend/uclapi/uclapi/settings.py [hunk 3]
```diff
@@ -187,15 +187,15 @@
 
 USE_TZ = False
 
+# set it to True when developing using webpack-serve
+# so that static files are fetched from http://localhost:8080/
+WEBPACK_SERVE = True
```
Comment (ChristopherHammond13) [inline backend/uclapi/uclapi/settings.py]: Where is this used in the code? I was going to convert it to an environment variable but I can't find the reference

Comment (anirudhpillai) [reply backend/uclapi/uclapi/settings.py]: Oh, don't need it anymore now. You can remove it.

**Pull request merged**